installation/uninstall_configmanagement.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script uninstalls ConfigManagement
+
+set -euox pipefail
+
+kubectl delete deployment -n config-management-system config-management-operator --ignore-not-found --cascade=foreground
```

Comment:
Before deleting the operator Deployment, should we disable HNC first to allow the operator to undeploy HNC resources gracefully?

If not in the uninstallation script, the upgrade instruction should have 3 steps:
1. disable HNC if it is enabled
2. uninstall the current operator
3. upgrade to the new version

Reply:
I'm assuming that the desired behavior when removing ACM operator is to leave subcomponents (HNC, PoCo, CS) orphaned on the cluster. Is there a reason the user would want to uninstall HNC during the procedure?

I don't think it needs to be part of the uninstallation script, but if there's a compelling use case then we could include the upgrade instructions that you suggest.

Reply:
For CS, we can safely leave it orphaned because we don't want to pause the reconciliation and the new installation will upgrade old reconcilers.
For PoCo, I am not sure about the migration process.
For HNC, users won't be able to disable it after the upgrade if we leave the subcomponent around. They will
a. continue using the orphaned and unsupported subcomponent without migration.
b. encounter controller fights with both orphaned HNC and OSS HNC.
c. manually delete each individual resource (controller, RBAC, service account, and etc) before installing OSS HNC.

To prevent the above issues, we should  include disabling HNC as part of uninstallation or as a prerequisite. Then users are forced to disable HNC  before uninstallation and upgrade.

Reply:
The next PR will include the installation tooling and instructions. Let's continue the discussion on that PR.

Reply:
The installation instruction may affect the uninstallation logic. We may either disable HNC, or block uninstallation if HNC is enabled.

Reply:
This script is intended to be the same behavior as `nomos migrate`, which does a simple uninstall that orphans all components.

It feels like the HNC specific logic is out of scope here. Is there a design doc for the experience of migrating to OSS HNC? It seems like a more consistent approach would be to orphan everything with the ACM uninstall and then have a separate tool that migrates from the legacy HNC to OSS HNC. I assume not all users want to migrate to OSS HNC and some might want to stay on the legacy HNC.

Reply:
Sent https://github.com/GoogleContainerTools/kpt-config-sync/pull/1454 to first update the behavior of `nomos migrate`.

Reply:
Updated the script to block if HNC is enabled